test/unit/lib/configuration/variables/sources/instance-dependent/get-sls.test.js

```diff
@@ -35,22 +41,42 @@ describe('test/unit/lib/configuration/variables/sources/instance-dependent/get-s
       options: {},
     });
     serverlessInstance.init();
+    const options = {};
+    if (typeof cliOptionStage === 'string') {
+      options.stage = cliOptionStage;
+    }
     await resolve({
       servicePath: process.cwd(),
       configuration,
       variablesMeta,
       sources: { self: selfSource, sls: getSlsSource(serverlessInstance) },
-      options: {},
+      options,
       fulfilledSources: new Set(['self', 'sls']),
     });
-  });
+  };
+
+  beforeEach(async () => await initializeServerless());
 
   it('should resolve instanceId', () => {
     if (variablesMeta.get('custom\0sls')) throw variablesMeta.get('custom\0sls').error;
     expect(typeof serverlessInstance.instanceId).to.equal('string');
     expect(configuration.custom.sls).to.equal(serverlessInstance.instanceId);
   });
 
+  it('should resolve ${sls:stage} to "dev" by default', () => {
```

Comment:
I would do one `it("should resolve ${sls:stage}")` and in it's context simply test one by one, all three variants